### ISSUE TITLE
Incorrect Time Zone in Blazor WASM

### ISSUE THREAD

**Issue by macriluis** [dotnet/runtime#40723]
### Describe the bug
Using a Windows PC, you set your Time Zone as (UTC-03:00) City of Buenos Aires
In Blazor WebAssembly client, the Time Zone is UTC.
### To Reproduce

Create  a new Blazor WebAssembly app.

In Index.razor add:

    Your Time: @DateTime.Now.ToString()
    Your Time Zone: @TimeZoneInfo.Local.DisplayName
<br />

In Control Panel or Configuration, set the Time Zone as (UTC-03:00) City of Buenos Aires
In Blazor WebAssembly client, the Time zone is UTC.
Wrong

In Control Panel or Configuration, set the Time Zone as (UTC-04:00) Santiago
In Blazor WebAssembly client, the Time zone is America/Santiago.
Correct

In Control Panel or Configuration, set the Time Zone as (UTC-03:00) Montevideo
In Blazor WebAssembly client, the Time zone is America/Montevideo.
Correct

### Further technical details
- ASP.NET Core version
- Include the output of `dotnet --info`
SDK de .NET Core (reflejando cualquier global.json):
 Version:   3.1.400
 Commit:    035fb2aa2f

Entorno de tiempo de ejecución:
 OS Name:     Windows
 OS Version:  10.0.18363
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\3.1.400\

Host (useful for support):
  Version: 3.1.6
  Commit:  3acd9b0cd1

.NET Core SDKs installed:
  3.1.300-preview-015115 [C:\Program Files\dotnet\sdk]
  3.1.400-preview-015151 [C:\Program Files\dotnet\sdk]
  3.1.400 [C:\Program Files\dotnet\sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.20 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.20 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.6 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.20 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.6 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.6 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

- The IDE (VS / VS Code/ VS4Mac) you're running on, and it's version
Visual Studio Community 2019 16.7.0

Tested with OSs: Windows 10, 7, Server 2012 R2, 2016, 2019. All in Spanish, I didn´t have the chance to test in other languages.
Tested with Browsers: Edge, Chrome, Opera, Firefox

![BuenosAires](https://user-images.githubusercontent.com/50157508/89927921-6fe8bb00-dbdd-11ea-8c4d-0da658049b38.jpg)
![Montevideo](https://user-images.githubusercontent.com/50157508/89927929-724b1500-dbdd-11ea-9daa-10f13cc015d1.jpg)
![Santiago](https://user-images.githubusercontent.com/50157508/89927933-74ad6f00-dbdd-11ea-8698-364cf02f08ce.jpg)

**Comment by captainsafia** [comment]
@lewing This user reported an issue with incomplete timezone data in 3.2. Transferring it to the runtime repo to confirm that the data is complete for .NET 5.

**Comment by tqiu8** [comment]
The correct name is now being outputted for .NET5

Code:
```
using System;
using System.Globalization;

public class Test
{
    public static void Main (String[] args) {
        TimeZoneInfo utc = TimeZoneInfo.FindSystemTimeZoneById("America/Argentina/Buenos_Aires");
        Console.WriteLine(utc.Id);
        Console.WriteLine (utc.DisplayName);
    }
}
```

Output:
```
../../../../../.dotnet/dotnet build --nologo -v:q -consoleloggerparameters:NoSummary /p:TargetArchitecture=wasm /p:TargetOS=Browser /p:Configuration=Release WasmSample.csprojcd bin/Release/publish && ~/.jsvu/v8 --expose_wasm runtime.js -- --run WasmSample.dll                                                                                         
Arguments: --run,WasmSample.dll                                                                                                                               
MONO_WASM: Loaded: System.Console.dll                                                                                                                
MONO_WASM: Loaded: System.Private.CoreLib.dll
MONO_WASM: Loaded: System.Private.Uri.dll                                                                                                                                     MONO_WASM: Loaded: System.Runtime.dll
MONO_WASM: Loaded: System.Runtime.InteropServices.dll
MONO_WASM: Loaded: System.Threading.dll
MONO_WASM: Loaded: WasmSample.dll
MONO_WASM: Loaded: icudt.dat 
MONO_WASM: Loaded: dotnet.timezones.blat 
MONO_WASM: Initializing mono runtime 
MONO_WASM: ICU data archive(s) loaded, disabling invariant mode 
console.debug: mono_wasm_runtime_ready fe00e07a-5519-4dfe-b35a-f867dbaf2e28 
Initializing.....  
America/Argentina/Buenos_Aires                                                                                                                                                
(UTC-03:00) -03  
```

**Comment by macriluis** [comment]
Thank You very much!!!
In Preview 8, the DateTime is correct.
Now there are missing some strings in:

Time Zone DisplayName: @TimeZoneInfo.Local.DisplayName
Time Zone StandardName: @TimeZoneInfo.Local.StandardName

![net5pv8](https://user-images.githubusercontent.com/50157508/91244786-9f421080-e723-11ea-8677-eb6f06621be9.jpg)

But, these for sure will be ok in the final release.

Thank you again, you solved me a big problem.

**Comment by marek-safar** [comment]
@EgorBo / @tqiu8 are the display names known issue?

**Comment by EgorBo** [comment]
@marek-safar I disabled naive time zone names (e.g. print "Московоское время" for Moscow timezone when current culture is ru-RU) but english name are expected to be fine... will take a look.

**Comment by EgorBo** [comment]
ok so english names are expected to be in the timezoneinfo binary file cc @tqiu8

**Comment by marek-safar** [comment]
@tqiu8 what is the update on this issue?

**Comment by tqiu8** [comment]
I don't think it's a problem on our side, but rather with how IANA abbreviates the standard time. This is what I have from their 2020a timezone file:

```
# From Paul Eggert (2013-09-05):
# Perhaps San Luis operates on the legal fiction that it is at -04
# with perpetual summer time, but ordinary usage typically seems to
# just say it's at -03; see, for example,
# http://es.wikipedia.org/wiki/Hora_oficial_argentina
# We've documented similar situations as being plain changes to
# standard time, so let's do that here too.  This does not change UTC
# offsets, only tm_isdst and the time zone abbreviations.  One minor
# plus is that this silences a zic complaint that there's no POSIX TZ
# setting for time stamps past 2038.

# From Paul Eggert (2013-02-21):
# Milne says Córdoba time was -4:16:48.2.  Round to the nearest second.

# Zone	NAME		GMTOFF	RULES	FORMAT	[UNTIL]
#
# Buenos Aires (BA), Capital Federal (CF),
Zone America/Argentina/Buenos_Aires -3:53:48 - LMT	1894 Oct 31
			-4:16:48 -	CMT	1920 May    # Córdoba Mean Time
			-4:00	-	-04	1930 Dec
			-4:00	Arg	-04/-03	1969 Oct  5
			-3:00	Arg	-03/-02	1999 Oct  3
			-4:00	Arg	-04/-03	2000 Mar  3
			-3:00	Arg	-03/-02
```

**Comment by macriluis** [comment]
@marek-safar now with RC1 the TimeZone Info is wrong again

![BsAsRC1](https://user-images.githubusercontent.com/50157508/93206106-44eb0d00-f72f-11ea-8d1c-cbd3e24de098.jpg)

**Comment by marek-safar** [comment]
@EgorBo any further thoughts on this?

**Comment by EgorBo** [comment]
@marek-safar my guess the timezone blat file is missing (old template?)

**Comment by macriluis** [comment]
The solution TTZ003 was created from scratch using Visual Studio 2019 16.8 Preview 3.

**Comment by macriluis** [comment]
@marek-safar with RC2 it's still wrong.
Demo app (TTZ4) was created with Visual Studio 2019 16.8 Preview 4
Should I open a new issue?

![net5rc2](https://user-images.githubusercontent.com/50157508/96156986-1dcb5b00-0ee8-11eb-8478-65d65e755906.jpg)

**Comment by lprada** [comment]
Any news, why is this close?

**Comment by lewing** [comment]
@lprada the issue with Buenos Aires (and others) has been fixed in main and will be released in 5.0.1
